From 6be8d7758d4ea113196e947100f37e629347c402 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Tue, 6 Jun 2023 10:50:22 +0200
Subject: [PATCH 1/4] Disable font access in Universal Windows Platform builds

EnumFontFamiliesExW/GetDC calls are not allowed to access the fonts and there
doesn't seem to be an equivalent call.
---
 src/libbluray/bdj/native/java_awt_BDFontMetrics.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/libbluray/bdj/native/java_awt_BDFontMetrics.c b/src/libbluray/bdj/native/java_awt_BDFontMetrics.c
index f0cd090a..be03fdb8 100644
--- a/src/libbluray/bdj/native/java_awt_BDFontMetrics.c
+++ b/src/libbluray/bdj/native/java_awt_BDFontMetrics.c
@@ -39,8 +39,11 @@
 #endif
 
 #if defined(_WIN32) && defined (HAVE_FT2)
+#include <winapifamily.h>
+#if WINAPI_FAMILY_PARTITION (WINAPI_PARTITION_DESKTOP) // EnumFontFamiliesExW/GetDC is not allowed in UWP
 #define NEED_WIN32_FONTS
 #endif
+#endif
 
 #ifdef NEED_WIN32_FONTS
 #include "file/dirs.h"  // win32_get_font_dir
-- 
2.37.3.windows.1

